File: frontend/app/api/transactions/maintenanceTypes/updateMaintenanceTypeTransaction.ts
import { API_BASE_URL } from "../../client/apiClient"
import { MaintenanceType } from "../../../types/maintenanceTypeTypes"
import { getMaintenanceTypesFromApi } from "../../maintenanceTypes/fetchMaintenanceTypes"
import { normalizeMaintenanceType,toUpdateMaintenanceTypeRequest } from "../../../utils/maintenanceTypeMapper"
import { authFetch } from "../../client/apiClient"

type UpdateMaintenanceTypeTransactionParams = {
                                                maintenanceType: MaintenanceType
                                                setMaintenanceTypes: any
                                              }

export async function updateMaintenanceTypeTransaction({
                                                         maintenanceType,
                                                         setMaintenanceTypes
                                                       }: UpdateMaintenanceTypeTransactionParams
                                                     )
{
  console.log("updateMaintenanceTypeTransaction")

  await fetch(
                `${API_BASE_URL}/update-maintenance-type`,
                {
                  method: "POST",
                  headers: {
                            "Content-Type":
                            "application/json"
                            },credentials: "include",
                  body: JSON.stringify(
                                          toUpdateMaintenanceTypeRequest(
                                                                            maintenanceType
                                                                          )
                                        )
                }
              )

  const maintenanceTypes =
    await getMaintenanceTypesFromApi()

  setMaintenanceTypes(
                        maintenanceTypes.map(
                                              normalizeMaintenanceType
                                            )
                      )
}

File: backend/transactions/maintenance_types/update_maintenance_type_transaction.py
from schemas.maintenance_type_schemas import (UpdateMaintenanceTypeRequest)
from maintenance_types.update_maintenance_type import (update_maintenance_type)
from supabase import Client

def update_maintenance_type_transaction(
                                            client:Client,
                                            maintenance_type: UpdateMaintenanceTypeRequest,
                                            hospital_id: str
                                        ):

    print("update_maintenance_type_transaction")

    return update_maintenance_type(
                                        client,
                                        maintenance_type,
                                        hospital_id
                                  )